src/widget/wtrackmenu.cpp

```diff
@@ -1595,6 +1625,161 @@ void WTrackMenu::slotClearAllMetadata() {
             &trackOperator);
 }
 
+namespace {
+
+class RemoveTrackFilesFromDiskTrackPointerOperation : public mixxx::TrackPointerOperation {
+  public:
+    mutable QList<TrackRef> tr_tracksToPurge;
+    mutable QList<QString> tr_tracksToKeep;
+
+  private:
+    void doApply(
+            const TrackPointer& pTrack) const override {
+        auto trackRef = TrackRef::fromFileInfo(
+                pTrack->getFileInfo(),
+                pTrack->getId());
+        VERIFY_OR_DEBUG_ASSERT(trackRef.isValid()) {
+            return;
+        }
+        QString location = pTrack->getLocation();
+        QFile file(location);
+        if (file.exists() && !file.remove()) {
```

Comment:
I have just realized that we have moveToTrash() since Qt 5.15 
https://doc.qt.io/qt-5/qfile.html#moveToTrash

This looks like a nice replacement here. What do you think?

Reply:
Considering that the deletion is revertable with that we should rather use that, indeed.
But: we agreed to support the latest Ubuntu LTS and the on before. 20.04 only provides Qt5.12, so we'd need to release after 2022-04-21?

Reply:
For me it would be OK to have a checkbox or something "Move to Trash" vs. "delete" only available for suitable Qt builds. 
The alternative would be moving Qt implementation to the lib folder. It are three one page functions one for each OS.

Reply:
> The alternative would be moving Qt implementation to the lib folder.

Hacky, but maybe it could work. I think that's beyond the scope of this PR though.

Reply:
Currently there are two safeguards:
* Delete is in a submenu
* There's a big confirmation dialog

If you insist to use a third for all who use Qt5.15+ I can make `remove`/`moveToTrash` conditional.

Reply:
I don't really insist. I think it is finally up to you if you have interest to go the extra mile.

The issue has two aspects, loosing the library entries and loosing the file. I am not sure if we can streamline the UI if we know that "loosing the file" can be undone.

I am currently consider to contribute a moveToTrash function for all our targets.

So if you wish .... :-)

Reply:
I have this implemented functionally but stashed away because atm I'm not keen on ifdef'ing and rephrasing dialogs texts, as well as building Qt 5.15 for testing.
Though, I'll finish this sooner or later.